src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java

```diff
@@ -254,9 +280,73 @@ public void rollback() {
         }
 
         @Override
-        public Pair<byte[], byte[]> keyRange(String table) {
-            // TODO: get first key and lastkey of fake tikv table
-            return null;
+        public List<Pair<byte[], byte[]>> keyRanges(String table) {
+            E.checkArgument(StringUtils.isNotBlank(table),
+                            "Table don't allow empty");
+
+            ByteString startKey = this.key(table);
+            ByteString endKey = ByteString.copyFrom(
+                                Key.toRawKey(startKey).nextPrefix().getBytes());
+            List<TiRegion> regions = this.fetchRegionsFromRange(startKey,
+                                                                endKey);
+            if (regions == null) {
+                return Collections.emptyList();
+            }
+
+            int tablePrefixLength = startKey.size();
+
+            return regions.stream()
+                          .filter((region) -> {
+                                 return !region.getStartKey().isEmpty();
+                           }).map((region) -> {
+                               /*
+                                * Tikv region look like this:
+                                * region1 [startKey1, regionEndKey1)
+                                * region2 [startKey2, regionEndKey2)
+                                *  ...
+                                * regionN [startKeyN, regionEndKeyN]
+                                * if regionEndKeyN < tableEndKey, the table real
+                                * end key is regionEndKeyN, otherwise the real
+                                * end key is table end key
+                                */
+                                 byte[] realEndKey = null;
+                                 if (FastByteComparisons.compareTo(
+                                     endKey.toByteArray(),
+                                     region.getEndKey().toByteArray()) < 0) {
```

Comment:
let region.getEndKey() first when compareTo()?
and can use `region.getEndKey().compareTo(endKey) > 0`

Reply:
The region type is TiRegion,  and the type is no the compareTo method